provider_client.go

```diff
@@ -146,15 +170,16 @@ func (client *ProviderClient) Request(method, url string, options RequestOpts) (
 		}
 	}
 
-	// Issue the request.
+	// Set user-agent header
+	req.Header.Set("User-Agent", client.UserAgent.Join())
```

Comment:
I would do this *before* `options.MoreHeaders` is processed, up above. That lets `MoreHeaders` be "the final word" on what's actually sent in the request, which in turn keeps us from running into the "OmitAccept" kinds of scenarios down the road.

Not that I can foresee us omitting the user agent from a specific request for any reason - in this case it would be more about consistency.